common/persistence/queueManager.go

```diff
@@ -68,7 +68,20 @@ func (q *queueManager) DeleteMessagesBefore(ctx context.Context, messageID int64
 }
 
 func (q *queueManager) UpdateAckLevel(ctx context.Context, messageID int64, clusterName string) error {
-	return q.persistence.UpdateAckLevel(ctx, messageID, clusterName)
+	ackLevels, err := q.persistence.GetAckLevels(ctx)
```

Comment:
Do we need to do it in a transaction?

Reply:
Yes. Let me update this one

Reply:
The persistence layer already does transaction updates. I move this verification logic one layer down.